Xplat/src/main/java/vazkii/botania/mixin/PistonBaseBlockMixin.java

```diff
@@ -29,10 +39,19 @@ private void preMoveBlocks(Level level, BlockPos pos, Direction dir, boolean ret
 		at = @At(value = "RETURN"),
 		method = "moveBlocks(Lnet/minecraft/world/level/Level;Lnet/minecraft/core/BlockPos;Lnet/minecraft/core/Direction;Z)Z"
 	)
-	private void postMoveBlocks(Level level, BlockPos pos, Direction dir, boolean retract,
+	private void postMoveBlocks(Level level, BlockPos pos, Direction dir, boolean extending,
 			CallbackInfoReturnable<Boolean> cir) {
 		if (!level.isClientSide()) {
 			EthicalTntHelper.endTrackingTntEntitiesAndCheck();
 		}
 	}
+
+	/**
+	 * When force-pushing, set the {@code extending} parameter to {@code false} right before the if block that creates
+	 * a moving piston head. This will skip all the code we don't want to execute.
+	 */
+	@ModifyVariable(method = "moveBlocks", at = @At(value = "LOAD", ordinal = 4), argsOnly = true)
```

Comment:
is there a way to further filter the At load, for example by type? The narrower you make it the less brittle this mixin is.